src/include/auth.h

```diff
@@ -141,7 +133,12 @@ typedef struct auth_def {
 	 * the function pointer to decrypt data
 	 */
 	int (*decrypt_data)(void *ctx, void *data_in, size_t len_in, void **data_out, size_t *len_out);
-} auth_def_t;
+
+	/*
+	 * pointer to next authdef structure
+	 */
+	auth_def_t *next;
```

Comment:
any particular reason why you are changing this from an array to a linked list?

Reply:
Yes, @agrawalravi90, earlier we were loading authdefs only which was listed in PBS_SUPPORTED_AUTH_METHODS, PBS_AUTH_METHOD, and PBS_ENCRYPT_METHOD (aka fixed set of authdefs), so I used array and now as I mentioned in the PR description about removing requirement listing PBS_ENCRYPT_METHOD of the client in PBS_SUPPORTED_AUTH_METHODS or PBS_ENCRYPT_METHOD on the server, even if server is not using them.

So now until the client connects with its selected PBS_ENCRYPT_METHOD we don't how what to load for authdef, and so unknown size of loaded authdefs...
+
the pointer to authdefs is bound to per connection struct (aka `pbs_tcp_chan_t->auths[]->def`, see dis.h), now if use array, and if new encrypt method comes which we need to load and add it to the array so we have to realloc() array which will change the pointers of authdef, which will lead bad pointer in connection struct

HTH!

Reply:
Got it, thanks for explaining